### ISSUE TITLE
last.ckpt and checkpoint_<epoch>.ckpt skip epochs whenever eval_interval > 1

### ISSUE THREAD

**Issue by raghav-rathi** [roboflow/rf-detr#1509]
### Search before asking

- [x] I have searched the RF-DETR issues and found no similar bug report.

### Bug

With `eval_interval > 1`, `last.ckpt` is only rewritten on the epochs that validate, and the `checkpoint_<epoch>.ckpt` archives skip every interval epoch that is not also an evaluation epoch.

`build_trainer` registers the two resume/archive `ModelCheckpoint` callbacks (`src/rfdetr/training/trainer.py:356-382` on `develop` @ 6b91a48e) without `save_on_train_epoch_end`, and sets `check_val_every_n_epoch=tc.eval_interval`. Lightning's `ModelCheckpoint._should_save_on_train_epoch_end` returns `False` whenever `check_val_every_n_epoch != 1`, so both callbacks only save from `on_validation_end`, which fires only every `eval_interval` epochs. The comment on the first callback says "overwritten every epoch".

Reproduced with a real `Trainer.fit()` over the test suite's mocked module (`epochs=6, eval_interval=2, checkpoint_interval=3`), watching `last.ckpt`'s mtime and the archive directory from `on_train_epoch_end`:

```
epoch 0: last.ckpt rewritten this epoch=n/a; archives=[]
epoch 1: last.ckpt rewritten this epoch=yes; archives=[]
epoch 2: last.ckpt rewritten this epoch=NO;  archives=[]      <- checkpoint_2.ckpt is due here
epoch 3: last.ckpt rewritten this epoch=yes; archives=[]
epoch 4: last.ckpt rewritten this epoch=NO;  archives=[]
epoch 5: last.ckpt rewritten this epoch=NO;  archives=[]
final files: ['checkpoint_5.ckpt', 'last.ckpt']
```

`checkpoint_5.ckpt` only exists because `_ForceLastEpochValidationCallback` forces a validation on the last epoch. Consequences: a crash between validations loses up to `eval_interval - 1` epochs of progress, and with the defaults (`checkpoint_interval=10`) plus, say, `eval_interval=3`, the archives for epochs 10, 20, 40, 50, 70, 80 never exist (only epochs that are multiples of both do). `BestModelCallback` is metric-driven and unaffected.

### Environment

- RF-DETR: 1.11.0.dev0 (`develop` @ 6b91a48e)
- OS: Ubuntu 22.04
- Python: 3.12
- PyTorch: 2.14.0+cu132, pytorch_lightning 2.6.6
- GPU: irrelevant (CPU repro)

### Minimal Reproducible Example

```python
from pathlib import Path
from unittest.mock import MagicMock, patch
from rfdetr.config import RFDETRBaseConfig, TrainConfig
from rfdetr.training import build_trainer
from rfdetr.training.module_data import RFDETRDataModule
from rfdetr.training.module_model import RFDETRModelModule
from tests.training.helpers import _fake_postprocess, _FakeCriterion, _FakeDataset, _make_param_dicts, _TinyModel

out = Path("out_cadence")
mc = RFDETRBaseConfig(pretrain_weights=None, device="cpu", num_classes=3)
tc = TrainConfig(dataset_dir="ds", output_dir=str(out), epochs=6, batch_size=2, num_workers=0,
                 eval_interval=2, checkpoint_interval=3, use_ema=False, tensorboard=False, run_test=False)
with (
    patch("rfdetr.training.module_model.build_model_from_config", return_value=_TinyModel()),
    patch("rfdetr.training.module_model.build_criterion_from_config", return_value=(_FakeCriterion(), MagicMock(side_effect=_fake_postprocess))),
    patch("rfdetr.training.module_data.build_dataset", return_value=_FakeDataset(length=20)),
    patch("rfdetr.training.module_model.get_param_dict", side_effect=lambda args, model: _make_param_dicts(model)),
):
    trainer = build_trainer(tc, mc, accelerator="cpu", limit_train_batches=2, limit_val_batches=2)
    trainer.fit(RFDETRModelModule(mc, tc), datamodule=RFDETRDataModule(mc, tc))
print(sorted(p.name for p in out.glob("*.ckpt")))   # ['checkpoint_5.ckpt', 'last.ckpt']; checkpoint_2.ckpt is missing
```

### Additional

Fix: pass `save_on_train_epoch_end=True` to both callbacks. Validation runs inside the training epoch before `on_train_epoch_end`, so a checkpoint written on an evaluation epoch still carries that epoch's validation results and callback state. I have this ready with a regression test that runs the configuration above and asserts `last.ckpt` holds epoch `N-1` at the start of every epoch `N` and that `checkpoint_2.ckpt` exists; PR follows.

### Are you willing to submit a PR?

- [x] Yes, I'd like to help by submitting a PR!
